src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -1341,13 +1342,41 @@ void MacroAssembler::ic_call(address entry, jint method_index) {
   RelocationHolder rh = virtual_call_Relocation::spec(pc(), method_index);
 #ifdef _LP64
   // Needs full 64-bit immediate for later patching.
-  mov64(rax, (intptr_t)Universe::non_oop_word());
+  mov64(rax, (int64_t)Universe::non_oop_word());
 #else
   movptr(rax, (intptr_t)Universe::non_oop_word());
 #endif
   call(AddressLiteral(entry, rh));
 }
 
+int MacroAssembler::ic_check_size() {
+  return LP64_ONLY(14) NOT_LP64(12);
+}
+
+int MacroAssembler::ic_check(int end_alignment) {
+  Register receiver = LP64_ONLY(j_rarg0) NOT_LP64(rcx);
+  Register data = rax;
+  Register temp = LP64_ONLY(rscratch1) NOT_LP64(rbx);
+
```

Comment:
It would be nice if VerifyOops check could be added back, maybe as a follow-up RFE?

Reply:
Good idea.